contracts/Market.sol

```diff
@@ -82,9 +103,35 @@ contract Market is IMarket, Ownable, Pausable {
         uint256 reservePrice,
         string adMetadataURI
     ) payable external whenNotPaused {
-        // @todo implement
+        address currentOwner = _token.ownerOf(tokenId);
 
-        assert(false);
+        // Timestamp of the current period starrt.
+        uint256 currentPeriodStart = (block.timestamp / _interval) * _interval;
+
+        uint256 nextPrice = tokenPrice[tokenId] + _epsilon;
+        uint256 tax = nextPrice * _taxRatePerInterval * numberOfIntervals;
```

Comment:
Ah, this of course it is broken. Need to divide by the precision of `_taxRatePerInterval`.

Better to have some "fixed point" helpers.